* To calculate the market risk of a stock, you can calculate its beta value. Beta measures the volatility of a stock relative to the overall market. A beta value of 1 indicates that the stock's volatility is equivalent to the market's volatility, while a beta value greater than 1 indicates that the stock is more volatile than the market, and a beta value less than 1 indicates that the stock is less volatile than the market.

In [1]:
import yfinance as yf


In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [3]:
tickers = ["AMZN"]

In [4]:
data = yf.download(tickers,start = "2015-01-01")
data

[*********************100%***********************]  1 of 1 completed


Open        High         Low       Close   Adj Close  \
Date                                                                     
2015-01-02   15.629000   15.737500   15.348000   15.426000   15.426000   
2015-01-05   15.350500   15.419000   15.042500   15.109500   15.109500   
2015-01-06   15.112000   15.150000   14.619000   14.764500   14.764500   
2015-01-07   14.875000   15.064000   14.766500   14.921000   14.921000   
2015-01-08   15.016000   15.157000   14.805500   15.023000   15.023000   
...                ...         ...         ...         ...         ...   
2023-04-17  103.160004  103.730003  101.589996  102.739998  102.739998   
2023-04-18  103.949997  104.199997  101.519997  102.300003  102.300003   
2023-04-19  101.580002  105.120003  101.389999  104.300003  104.300003   
2023-04-20  103.529999  105.250000  103.209999  103.809998  103.809998   
2023-04-21  106.099998  108.150002  105.080002  106.959999  106.959999   

              Volume  
Date                  
2015-01-02  55664000  
2015-01-05  55484000  
2015-01-06  70380000  
2015-01-07  52806000  
2015-01-08  61768000  
...              ...  
2023-04-17  39919500  
2023-04-18  39790500  
2023-04-19  58398900  
2023-04-20  57696900  
2023-04-21  86689700  

[2090 rows x 6 columns]

In [5]:
start_date = '2015-01-01'
end_date  = pd.Timestamp.today().strftime('%Y-%M-%d')

In [6]:
# Calculate the daily returns
daily_returns = data['Adj Close'].pct_change()

In [7]:
# Get the S&P 500 index data
tickers = ["GOOGL"]
data_google = yf.download(tickers,start = "2015-01-01")
data_google

[*********************100%***********************]  1 of 1 completed


Open        High         Low       Close   Adj Close  \
Date                                                                     
2015-01-02   26.629999   26.790001   26.393999   26.477501   26.477501   
2015-01-05   26.357500   26.399500   25.887501   25.973000   25.973000   
2015-01-06   26.025000   26.060499   25.277500   25.332001   25.332001   
2015-01-07   25.547501   25.574499   25.182501   25.257500   25.257500   
2015-01-08   25.075500   25.375000   24.750999   25.345501   25.345501   
...                ...         ...         ...         ...         ...   
2023-04-17  104.660004  106.160004  104.519997  105.970001  105.970001   
2023-04-18  106.489998  106.540001  104.070000  104.500000  104.500000   
2023-04-19  103.580002  104.980003  103.070000  104.180000  104.180000   
2023-04-20  103.910004  106.250000  103.870003  105.290001  105.290001   
2023-04-21  105.470001  106.000000  104.779999  105.410004  105.410004   

              Volume  
Date                  
2015-01-02  26480000  
2015-01-05  41182000  
2015-01-06  54456000  
2015-01-07  46918000  
2015-01-08  73054000  
...              ...  
2023-04-17  37571200  
2023-04-18  26596400  
2023-04-19  20905700  
2023-04-20  27820800  
2023-04-21  25784600  

[2090 rows x 6 columns]

In [8]:
# Calculate the daily returns of the index
index_returns = data_google['Adj Close'].pct_change()

In [9]:
#Merge the stock and index returns into a single DataFrame
returns = pd.concat([daily_returns, index_returns], axis=1, join='inner')


In [10]:
# Calculate the covariance matrix and beta value
covariance_matrix = returns.cov()
beta = covariance_matrix.iloc[0, 1] / covariance_matrix.iloc[1, 1]

In [11]:
# Print the beta value
print(f'Beta of Amazon: {beta:.2f}')

Beta of Amazon: 0.77


* A beta of 0.77 means that the stock of Amazon is 23% less volatile than the overall market.

